cmd/kube-scheduler/app/server.go

```diff
@@ -185,13 +185,20 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
 	}
 
 	// Start up the healthz server.
+	var gracefulShutdownSecureServer func()
 	if cc.SecureServing != nil {
 		handler := buildHandlerChain(newHealthzAndMetricsHandler(&cc.ComponentConfig, cc.InformerFactory, isLeader, checks...), cc.Authentication.Authenticator, cc.Authorization.Authorizer)
-		// TODO: handle stoppedCh and listenerStoppedCh returned by c.SecureServing.Serve
-		if _, _, err := cc.SecureServing.Serve(handler, 0, ctx.Done()); err != nil {
+		stoppedCh, listenerStoppedCh, err := cc.SecureServing.Serve(handler, 0, ctx.Done())
```

Comment:
ok, this is passing a 0 timeout and the same context signal as closing channel to the `SecureServing.Serve` method, that IIRC the Server code means there is no gracefully termination and it closes it inmidiatly.

IIUIC the purpose of the PR, we want to have some gracefult termination to allow the existing connections to finish, for that it will require to have a different close channel than ctx.Done() and a greater timeout passed

Reply:
I don't know how long you want to wait, but I suggest to define a const or a variable
```suggestion
                 shutdownTimeout := 5 * time.Second
		stoppedCh, listenerStoppedCh, err := cc.SecureServing.Serve(handler, shutdownTimeout, ctx.Done())
```

Reply:
@togettoyou can you update as suggested here or we'll not wait for the connections closed.

Reply:
you have more information on the logic here

https://github.com/kubernetes/kubernetes/blob/70af917493983f1d8b1332f8dd0c03800c0a7f15/staging/src/k8s.io/apiserver/pkg/server/genericapiserver_graceful_termination_test.go#L124-L174